### PR TITLE
Switch migrations + web assets to Go 1.16 embedding

### PR DESCRIPTION
**Pull request by vito**
- tidier error check (thanks Go 1.16!)
- bump minimum Go to 1.16
- web: behavior: use go:embed, re-parse index.html
- migration: use go:embed
- remove packr

edit: woops this was created with `gh pr create -f` and it nixed all of the template. but this shouldn't really have any external impact at least. 🤔 further info available in the commit messages!

*note: this does not include Dex, which currently uses `pkger`*

### CODE REVIEW

**Comment by xtremerui** [comment]
This will be backported to release/6.7.x and release/7.0.x branches.

File: tsa/tsacmd/server.go [hunk 1]
```diff
@@ -86,7 +87,7 @@ func (server *server) Serve(listener net.Listener) {
 	for {
 		c, err := listener.Accept()
 		if err != nil {
-			if !strings.Contains(err.Error(), "use of closed network connection") {
+			if !errors.Is(err, net.ErrClosed) {
```
Comment (chenbh) [inline tsa/tsacmd/server.go:90]: Should also be applied to:
```
atc/worker/gclient/retryable_garden_connection.go
83:		if strings.Contains(err.Error(), "use of closed network connection") {
101:		if strings.Contains(err.Error(), "use of closed network connection") {
```

File: web/cmd/main.go [hunk 2]
```diff
@@ -1,40 +0,0 @@
...
-
-	http.Handle("/api/", proxyHandler)
-	http.Handle("/auth/", proxyHandler)
-	http.Handle("/oauth/", proxyHandler)
-	http.Handle("/", webHandler)
-
-	if err = http.ListenAndServe(":8081", nil); err != nil {
-		logger.Error("server-error", err)
-	}
-}
```
Comment (chenbh) [inline web/cmd/main.go:40]: What was the purpose of this file? Was it possible to build the UI server by itself?

**Review by taylorsilva — COMMENTED** [review]
LGTM outside of Daniel's one comment.

Everything loaded fine and didn't notice any errors in firefox's dev console. The embed package is pretty cool!

File: web/cmd/main.go (hunk 2 above)
Comment (taylorsilva) [reply web/cmd/main.go:40]: Yup https://github.com/concourse/web/blob/8c95285e65efc043abcf6d285b011c01d309087b/cmd/main.go

Comment (vito) [reply web/cmd/main.go:40]: Yeah, I don't think we use it anymore now that the `docker-compose` flow is pretty smooth so I figured I'd just remove it.

File: go.mod [hunk 3]
```diff
@@ -105,6 +104,6 @@ require (
 	sigs.k8s.io/yaml v1.2.0
 )
 
-go 1.13
+go 1.16
```
Comment (ringods) [inline go.mod:107]: @vito don't forget to update the Go version here as well:

https://github.com/concourse/concourse/blob/master/CONTRIBUTING.md#development-dependencies

Comment (vito) [reply go.mod:107]: good call!

**Pull request merged**